pkg/routeagent/controllers/route/route.go

```diff
@@ -526,9 +534,8 @@ func (r *Controller) handleRemovedEndpoint(obj interface{}) {
 			klog.Errorf("Could not convert object tombstone %v to an Endpoint", tombstone.Obj)
 			return
 		}
-		klog.V(4).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
 	}
-	klog.V(4).Infof("Informed of removed endpoint for route controller object: %v", object)
+	klog.V(log.DEBUG_LEVEL).Infof("Informed of removed endpoint for route controller object: %v", object)
```

Comment:
Can you include the following check with a TODO (we can handle it properly in a separate PR)
if object.Spec.ClusterID != r.clusterID {
   // TODO: Handle a remote endpoint removal use-case 
   return
}

CC: @tpantelis

Reply:
What would we need to handle in that case, so I can include it as part of the comments.

Reply:
Oh right, that's a good one.